### PR TITLE
BOSA21Q1-745 Dialogue citoyen - Locale not correctly set when sign up with CSAM

### PR DESCRIPTION
**Pull request by sergei-krylov**
1. Refactoring:
- move changes from bosa-cities extends (`master-national-survey` branch) to the `decidim-verifications_omniauth` module.
- related tickets:
  - BOSA21Q1-680 Dialogue Citoyen CSAM Login - wrong email entered at sign up. Allow user to reset his email after CSAM registration (before confirmation).
  - BOSA21Q1-682 Dialogue Citoyen - On register confirmation email received twice.

Commit in module: https://github.com/belighted/decidim-module-verifications_omniauth/commit/34e5d008c212120b1f34b659fb2b945cfa2704ba

2. BOSA21Q1-745 Dialogue citoyen - Locale not correctly set when sign up with CSAM

Commit in module: https://github.com/belighted/decidim-module-verifications_omniauth/commit/1bff9037e8865e73133cd8fcb2e10c0720782a85

### CODE REVIEW

**Comment by sergei-krylov** [comment]
@simon-lejeune , as I mentioned in the description - there are 2 changes via 2 separate commits.

First one is a refactoring to move the changes from extend to the module. I should have done this initially, but I didn't think that there would be more changes in the omniauth module.
No code (logic) changes made here, only moving the code.

2nd commit is the actual fix for ticket 745: https://github.com/belighted/decidim-module-verifications_omniauth/commit/1bff9037e8865e73133cd8fcb2e10c0720782a85, the changes are relatively small.

I can keep it as it was before, but to add the fix I would still need to add part of the omniauth module to the extend (to be precise, this method: https://github.com/belighted/decidim-module-verifications_omniauth/blob/1bff9037e8865e73133cd8fcb2e10c0720782a85/lib/extends/verifications_omniauth/decidim-core/app/commands/decidim/create_omniauth_registration.rb#L53-L103), so it will be bigger amount of code related to fix for 745.

File: Gemfile.lock [hunk 1]
```diff
@@ -523,8 +523,9 @@ GEM
       railties
       sprockets-rails
     graphql (1.12.6)
-    gyoku (1.3.1)
+    gyoku (1.4.0)
```
[the remark this answers is not in the document]
Comment (sergei-krylov) [reply Gemfile.lock:526]: As a result of `bundle install` :smile:

**Comment by ThomasCollignon** [comment]
I'm not against the refactoring, it enhances the readability for newbies like me. And I assume that if it's not done now it has few chances to be done it later.

**Comment by simon-lejeune** [comment]
> I'm not against the refactoring, it enhances the readability for newbies like me. And I assume that if it's not done now it has few chances to be done it later.

it won't be done later because it's not necessary, starting from the 7th of june the login will be completely disabled. the issue is only for national survey and the national survey branches here and in the verifications repo are really temporary branches, we can focus on making the simpler and, especially, safer patches (for example avoiding those dependencies changes). Personally I'm more in favour of the second pull request, but still trying to find an even simpler way to implement.

**Comment by sergei-krylov** [comment]
Closed in favor of https://github.com/belighted/bosa-cities/pull/102

**Pull request closed without merging**